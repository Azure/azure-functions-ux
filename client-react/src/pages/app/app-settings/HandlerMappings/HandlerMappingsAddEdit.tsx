import * as React from 'react';
import { TextField } from 'office-ui-fabric-react/lib-commonjs/TextField';
import { HandlerMapping } from '../../../../models/WebAppModels';
export interface HandlerMappingAddEditProps extends HandlerMapping {
  updateHandlerMapping: (item: HandlerMapping) => any;
}

const HandlerMappingsAddEdit: React.SFC<HandlerMappingAddEditProps> = props => {
  const { updateHandlerMapping, children, ...handlerMapping } = props;
  const updateHandlerMappingExtension = (extension: string) => {
    updateHandlerMapping({ ...handlerMapping, extension });
  };

  const updateHandlerMappingScriptProccessor = (scriptProcessor: string) => {
    updateHandlerMapping({ ...handlerMapping, scriptProcessor });
  };

  const updateHandlerMappingArguments = (argumentsVal: string) => {
    updateHandlerMapping({
      ...handlerMapping,
      arguments: argumentsVal,
    });
  };
  return (
    <div>
      <TextField label="Extension" id="extension" value={handlerMapping.extension} onChanged={updateHandlerMappingExtension} />
      <TextField
        label="Script Processor"
        id="value"
        value={handlerMapping.scriptProcessor}
        onChanged={updateHandlerMappingScriptProccessor}
      />
      <TextField label="Arguments" id="type" value={handlerMapping.arguments} onChanged={updateHandlerMappingArguments} />
    </div>
  );
};
export default HandlerMappingsAddEdit;
