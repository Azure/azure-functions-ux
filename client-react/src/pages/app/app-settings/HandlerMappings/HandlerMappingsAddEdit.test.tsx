import { shallow } from 'enzyme';
import * as React from 'react';
import HandlerMappingsAddEdit from './HandlerMappingsAddEdit';
const HandlerMappingsAddEditAny = HandlerMappingsAddEdit as any;
describe('Handler Mappings Add Edit', () => {
  describe('Shallow Testing', () => {
    let container;

    beforeEach(() => {
      container = shallow(<HandlerMappingsAddEditAny />);
    });

    it('renders without crashing', () => {
      expect(container.length).toBe(1);
    });

    it('render matches snapshot', () => {
      expect(container).toMatchSnapshot('DefaultSnapshot');
    });
  });
});
