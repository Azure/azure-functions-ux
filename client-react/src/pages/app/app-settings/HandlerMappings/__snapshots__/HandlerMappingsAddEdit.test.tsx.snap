// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Handler Mappings Add Edit Shallow Testing render matches snapshot: DefaultSnapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HandlerMappingsAddEdit />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <StyledTextFieldBase
          id="extension"
          label="Extension"
          onChanged={[Function]}
          value={undefined}
        />,
        <StyledTextFieldBase
          id="value"
          label="Script Processor"
          onChanged={[Function]}
          value={undefined}
        />,
        <StyledTextFieldBase
          id="type"
          label="Arguments"
          onChanged={[Function]}
          value={undefined}
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "id": "extension",
          "label": "Extension",
          "onChanged": [Function],
          "value": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "id": "value",
          "label": "Script Processor",
          "onChanged": [Function],
          "value": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "id": "type",
          "label": "Arguments",
          "onChanged": [Function],
          "value": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <StyledTextFieldBase
            id="extension"
            label="Extension"
            onChanged={[Function]}
            value={undefined}
          />,
          <StyledTextFieldBase
            id="value"
            label="Script Processor"
            onChanged={[Function]}
            value={undefined}
          />,
          <StyledTextFieldBase
            id="type"
            label="Arguments"
            onChanged={[Function]}
            value={undefined}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "id": "extension",
            "label": "Extension",
            "onChanged": [Function],
            "value": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "id": "value",
            "label": "Script Processor",
            "onChanged": [Function],
            "value": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "id": "type",
            "label": "Arguments",
            "onChanged": [Function],
            "value": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
